README.rst

```diff
@@ -0,0 +1,793 @@
+    RPM.
+  - `arch` is the `qualifiers` `key` for a package architecture
+  - Examples::
+
+        rpm:fedora/curl@7.50.3-1.fc25?arch=i386&distro=fedora-25
+        rpm:opensuse/curl@7.56.1-1.1.?arch=i386&distro=opensuse-tumbleweed
+
+
+Other candidate types to define:
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+- `alpine` for Alpine Linux apk packages:
+- `apache` for Apache projects packages:
+- `android` for Android apk packages:
+- `arch` for Arch Linux packages:
+- `atom` for Atom packages:
+- `bower` for Bower JavaScript packages:
+- `brew` for Homebrew packages:
+- `buildroot` for Buildroot packages
+- `cargo` for Rust packages:
+- `carthage` for Cocoapods Cocoa packages:
+- `chef` for Chef packages:
+- `clojars` for Clojure packages:
+- `cocoapods` for Cocoapods iOS packages:
+- `conan` for Conan C/C++ packages:
+- `coreos` for CoreOS packages:
+- `cpan` for CPAN Perl packages:
+- `cran` for CRAN R packages:
+- `ctan` for CTAN TeX packages:
+- `crystal` for Crystal Shards packages:
+- `drupal` for Drupal packages:
+- `dtype` for DefinitelyTyped TypeScript type definitions:
+- `dub` for D packages:
+- `elm` for Elm packages:
+- `eclipse` for Eclipse projects packages:
+- `gitea` for Gitea-based packages:
+- `gitlab` for Gitlab-based packages:
+- `guix` for Guix packages:
+- `hackage` for Haskell packages:
+- `haxe` for Haxe packages:
+- `hex` for Erlang and Elixir packages
+- `julia` for Julia packages:
+- `lua` for LuaRocks packages:
+- `melpa` for Emacs packages
+- `meteor` for Meteor JavaScript packages:
+- `nim` for Nim packages:
+- `nix` for Nixos packages:
+- `opam` for OCaml packages:
+- `openwrt` for OpenWRT packages:
+- `osgi` for OSGi bundle packages:
+- `p2` for Eclipse p2 packages:
+- `pear` for Pear PHP packages:
+- `pecl` for PECL PHP packages:
+- `perl6` for Perl 6 module packages:
+- `platformio` for PlatformIO packages:
+- `ebuild` for Gentoo Linux portage packages:
+- `pub` for Dart packages:
+- `puppet` for Puppet Forge packages:
+- `sourceforge` for Sourceforge-based packages:
+- `sublime` for Sublime packages:
+- `swift` for Swift packages:
+- `vim` for Vim scripts packages:
+- `wordpress` for Wordpress packages:
+- `yocto` for Yocto recipe packages
+
+
+Known `qualifiers` key/value pairs
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Note: Do not abuse of `qualifiers`: it can be tempting to use many qualifier
+keys but their usage should be limited to the bare minimum for proper package
+identification to ensure that a `purl` stays compact and readable in most cases.
+
+Additional, separate external attributes stored outside of a `purl` are the
+preferred mechanism to convey extra long and optional information such as a
+download URL, vcs URL or checksums in an API, database or web form.
+
+
+With this warning, the known `key` and `value` defined here are valid for use in
+all package types:
+
+- `repository_url` is an extra URL for an alternative, non-default package
```

Comment:
"non-default" leaves an open question about what default actually is.

Reply:
I added an explanation. This default is listed for a given type. such as packagist.org for `composer`